src/main/java/net/sf/jabref/remote/DBProcessor.java

```diff
@@ -0,0 +1,534 @@
+        executeUpdate(query);
+        LOGGER.info("SQL DELETE: " + query);
+        normalizeEntryTable();
+    }
+
+    /**
+     *  Prepares the database table for a new {@link BibEntry}.
+     *  Learning table structure: Columns which are not available are going to be created.
+     *
+     *  @param bibEntry Entry which pretends missing columns which should be created.
+     *
+     */
+    public void prepareEntryTableStructure(BibEntry bibEntry) {
+        Set<String> fieldNames = dbHelper.allToUpperCase(bibEntry.getFieldNames());
+        fieldNames.removeAll(dbHelper.allToUpperCase(dbHelper.getColumnNames(escape(ENTRY))));
+
+        String columnType = dbType == DBType.ORACLE ? " CLOB NULL" : " TEXT NULL DEFAULT NULL";
+
+        for (String fieldName : fieldNames) {
+            executeUpdate("ALTER TABLE " + escape(ENTRY) + " ADD " + escape(fieldName) + columnType);
+        }
+    }
+
+    /**
+     *  Deletes all unused columns where every entry has a value NULL.
+     */
+    public void normalizeEntryTable() {
+        ArrayList<String> columnsToRemove = new ArrayList<>();
+
+        columnsToRemove.addAll(dbHelper.allToUpperCase(dbHelper.getColumnNames(escape(ENTRY))));
+        columnsToRemove.remove(ENTRY_REMOTE_ID); // essential column
+        columnsToRemove.remove(ENTRY_ENTRYTYPE); // essential column
+
+        try (ResultSet resultSet = dbHelper.query("SELECT * FROM " + escape(ENTRY))) {
+            while (resultSet.next()) {
+                for (int i = 0; i < columnsToRemove.size(); i++) {
+                    if (resultSet.getObject(columnsToRemove.get(i)) != null) {
+                        columnsToRemove.remove(i);
+                        i--; // due to index shift
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            LOGGER.error("SQL Error: ", e);
+        }
+
+        String columnExpression = "";
+        String expressionPrefix = "";
+        if ((dbType == dbType.MYSQL) || (dbType == dbType.POSTGRESQL)) {
+            expressionPrefix = "DROP ";
+        }
+
+        for (int i = 0; i < columnsToRemove.size(); i++) {
+            String column = columnsToRemove.get(i);
+            columnExpression = columnExpression + expressionPrefix + escape(column);
+            columnExpression = i < (columnsToRemove.size() - 1) ? columnExpression + ", " : columnExpression;
+        }
+
+        if (dbType == dbType.ORACLE) {
+            columnExpression = "DROP (" + columnExpression + ")"; // DROP command in Oracle differs from the other systems.
+        }
+
+        if (columnsToRemove.size() > 0) {
+            executeUpdate("ALTER TABLE " + escape(ENTRY) + " " + columnExpression);
+        }
+    }
+
+
+    /**
+     * Converts all remotely present bib entries to the List of real {@link BibEntry} objects and retrieves them.
+     */
+    public List<BibEntry> getRemoteEntries() {
+        ArrayList<BibEntry> remoteEntries = new ArrayList<>();
```

Comment:
Yep, should be List<BibEntry> remoteEntries = new ArrayList<>()
Always use the generic List Interface where possible.